cpp/src/arrow/array/array_binary.h

```diff
@@ -217,6 +218,75 @@ class ARROW_EXPORT LargeStringArray : public LargeBinaryArray {
   Status ValidateUTF8() const;
 };
 
+// ----------------------------------------------------------------------
+// BinaryView and StringView
+
+/// Concrete Array class for variable-size binary view data using the
+/// BinaryViewType::c_type struct to reference in-line or out-of-line string values
+class ARROW_EXPORT BinaryViewArray : public PrimitiveArray {
```

Comment:
It's a bit weird for this to inherit from `PrimitiveArray` (see hierarchy in [1]). Could it extends `FlatArray` like `BinaryArray` and `StringArray` do?

[1] https://gist.github.com/felipecrv/3c02f3784221d946dec1b031c6d400db

Reply:
All the types that inherit from `PrimitiveArray` evaluate to `true` on the `is_primitive(type)` predicate.